lib/Doctrine/ORM/NoResultException.php

```diff
@@ -29,9 +29,15 @@ class NoResultException extends UnexpectedResultException
 {
     /**
      * Constructor.
+     * @param string     $message  [optional] The Exception message to throw.
+     * @param integer    $code     [optional] The Exception code.
+     * @param \Exception $previous [optional] The previous exception used for the exception chaining. Since 5.3.0
      */
-    public function __construct()
+    public function __construct($message = '', $code = 0, \Exception $previous = null)
```

Comment:
Why not just add `$message = 'No result was found for query although at least one row was expected.',` here instead of checks in the constructor body?